app/src/main/java/com/dessalines/thumbkey/db/AppDb.kt

```diff
@@ -613,8 +635,41 @@ val MIGRATION_16_17 =
         }
     }
 
+val MIGRATION_17_18 =
+    object : Migration(17, 18) {
+        override fun migrate(db: SupportSQLiteDatabase) {
+            db.execSQL(
+                "ALTER TABLE AppSettings ADD COLUMN auto_size_keys INTEGER NOT NULL DEFAULT $DEFAULT_AUTO_SIZE_KEYS",
+            )
+            // The default for this new setting is *enabled*, but that would be a change for existing users, so if
+            // migrating from a previous dabase version withouth the setting at all, assume the user manually set
+            // the actual size they want to use, hence defaulting legacy users to *disabled* on upgrade.
+            db.execSQL(
+                "UPDATE AppSettings SET auto_size_keys = 0",
```

Comment:
This is going to run no matter what, so you should pry remove it. Just make auto key size default true for now I spose.

Reply:
What do you mean "no matter what"? Won't this block of migrations only run if there is an existing version 17 database present?

Reply:
I'm saying you might as well use SQL's column defaults when adding a new column. 

There's no reason to do 2 steps of : create the column, then update it's value in a migration, when you can just do a single step of creating the column with a default.